tfx_addons/firebase_publisher/README.md

```diff
@@ -21,25 +21,51 @@ This project helps users to publish trained models directly from TFX Pusher comp
+  model_blessing: Optional[types.BaseChannel] = None,
+)
+```
+
+- Each inputs:
+  - `model` : the model from the upstream TFX component such as [Trainer](https://www.tensorflow.org/tfx/api_docs/python/tfx/v1/components/Trainer)
+  - `model_blessing` : the output of `blessing` from the Evaluator component to indicate if the given `model` is good enough to be pushed
+  - `display_name` : display name to appear in Firebase ML. This should be a unique value since it will be used to search a existing model to update
+  - `tags` : tags to appear in Firebase ML
+  - `storage_bucket` : GCS bucket where the hosted model is stored. `gs://` should not be included
+  - `credential_path` : an optional parameter, and it indicates GCS or local location where a Service Account Key (JSON) file is stored. If this parameter is not given, [Application Default Credentials](https://cloud.google.com/docs/authentication/production) will be used in GCP environment
+  - `options` : additional configurations to be passed to initialize Firebase app
+
+- It outputs the following information by the method [`_MarkPushed`](https://github.com/tensorflow/tfx/blob/3b5290aa77c2df52a4791715cfd761be7696fe81/tfx/components/pusher/executor.py#L222) from Pusher component
+  - `pushed` : indicator if the model is pushed without any issue or if the model is blessed.
+  - `pushed_destination` : URL string for easy access to the model from Firebase Console such as `f"https://console.firebase.google.com/u/1/project/{PROJECT_ID}/ml/custom"`
+  - `pushed_version` : version string of the pushed model. This is determined in the same manner as Pusher by `str(int(time.time()))`
+
+- The detailed behaviour of this component
+  - Initialize Firebase App with `firebase_admin.initialize_app` from [Firebase Admin SDK](https://firebase.google.com/docs/admin/setup). When `credential_path` is given, it first downloads the credential file and uses it in the `creds` argument of the `initialize_app()`. When `options` parameter of this component is not `None`, it will be passed to the `options` argument of the `initialize_app()`.
+  
+  - Download the model from the upstream TFX component if the model is blessed. Unfortunately, Firebase ML only lets us upload/host a model from the local storage, so this step is required. Along the way, if the model is `TFLite` format, local flag `is_tfile` will be marked as `True`
+  
+  - If the model is `SavedModel` (which can be determined if `is_tflite` is set to `False`), `TFLiteGCSModelSource.from_saved_model(model_path)` is called. This function will convert `SavedModel` to `TFLite` and stores it in the GCS bucket specified in `storage_bucket`. Otherwise, `TFLiteGCSModelSource.from_tflite_model_file(model_path)` is used to directly upload the given `TFLite` model file
+
+  - Search the list of models whose name is same to the `display_name`. If the list is empty, a new model will be created and hosted. If the list is non-empty, the existing modell will be updated. 
```

Comment:
I am unsure why someone would host a SavedModel on Firebase ML. Isn't a GCS bucket better suited for that? 

For TFLite models being hosted on Firebase, the Firebase SDK allow easy integration of them with mobile apps.

Reply:
To clarify, Firebase doesn't host `SavedModel`, but it converts `SavedModel` into TFLite format under the hood. `from_saved_model()` function does that. I guess it simply calls TensorFlow APIs to do this, but it let us concentrate on the logics related to Firebase thing.

Reply:
Ah I see. Then might be worth clarifying it in the proposal.